nav2_behavior_tree/include/nav2_behavior_tree/bt_service_node.hpp

```diff
@@ -240,6 +247,9 @@ class BtServiceNode : public BT::ActionNodeBase
   std::shared_future<typename ServiceT::Response::SharedPtr> future_result_;
   bool request_sent_{false};
   rclcpp::Time sent_time_;
+
+  // Can be set in on_tick or on_wait_for_result to indicate if a request should be sent.
+  bool should_send_request_;
```

Comment:
Would you rather this be called similarly as in the BTActionNode (should_send_goal_) or better it be semantically more correct?